src/Objs/Job/JobComponent.js

```diff
@@ -1,17 +1,106 @@
+import formatDate from 'utils/formatDate';
+import InPlaceEditingPlaceholder from 'Components/InPlaceEditingPlaceholder';
+import SchemaDotOrg from 'Components/SchemaDotOrg';
+
 Scrivito.provideComponent('Job', ({ page }) => {
   return (
     <div>
       <section className="bg-white">
         <div className="container">
+          <Date text="" date={ page.get('datePosted') } />
           <Scrivito.ContentTag tag="h1" className="h2" content={ page } attribute="title" />
-          <h2 className="h4">
-            <i className="fa fa-map-marker fa-lg" aria-hidden="true" title="location" />
-            { ' ' }
-            <Scrivito.ContentTag tag="span" content={ page } attribute="location" />
-          </h2>
+          <JobLocation job={ page } />
+          <JobValidThrough job={ page } />
+          <JobEmploymentTypes employmentTypes={ page.get('employmentType') }/>
         </div>
       </section>
       <Scrivito.ContentTag tag="div" content={ page } attribute="body" />
+
+      <SchemaDotOrg content={ page } />
     </div>
   );
 });
+
+function Date({ date, text }) {
+  if (!date) {
+    return (
+      <InPlaceEditingPlaceholder>
+        Select a date in the job page properties.
+      </InPlaceEditingPlaceholder>
+    );
+  }
+
+  return <span>{ text } { formatDate(date, 'mm/dd/yyyy') }</span>;
+}
+
+const JobValidThrough = Scrivito.connect(({ job }) => {
```

Comment:
Please exit the `JobValidThrough` with a `InPlaceEditingPlaceholder` is no `validThrough` is given (similar to `JobEmploymentTypes`). Also it's probablly better to not abstract `Date`, but inline it instead.